app/modules/tmr.c

```diff
@@ -108,7 +108,13 @@ static void alarm_timer_common(void* arg){
 	}else if(tmr->mode == TIMER_MODE_SEMI){
 		tmr->mode |= TIMER_IDLE_FLAG;
 	}
-	lua_call(L, 0, 0);
+  if (tmr->self_ref == LUA_REFNIL) {
+    uint32_t id = tmr - alarm_timers;
+    lua_pushinteger(L, id);
+  } else {
+    lua_rawgeti(L, LUA_REGISTRYINDEX, tmr->self_ref);
+  }
+	lua_call(L, 1, 0);
```

Comment:
Please stick to a consistent indentation style.
If unsure, follow the Lua style: 2 space indent + 1TBS + no trailing spaces.